subsys/bluetooth/controller/Kconfig

```diff
@@ -433,28 +438,68 @@ endif # BT_CONN
 
 config BT_CTLR_CHAN_SEL_2
 	bool "Channel Selection Algorithm #2"
-	depends on (BT_CONN || BT_CTLR_ADV_EXT) && BT_CTLR_CHAN_SEL_2_SUPPORT
+	depends on (BT_CONN || BT_CTLR_ADV_PERIODIC) && BT_CTLR_CHAN_SEL_2_SUPPORT
```

Comment:
BT_CONN is included as per the spec requirementin v5.2, Vol6. Part B, section 4.6.12.
"A Controller that supports connections shall also support the Channel Selection Algorithm #2 feature."

When support for connection over extended advertising is added, it will enforce "select".